awxkit/awxkit/api/pages/projects.py

```diff
@@ -43,6 +43,16 @@ def payload(self, organization, scm_type='git', **kwargs):
             'allow_override')
         update_payload(payload, fields, kwargs)
 
+        for fk_field in ('execution_environment', 'default_environment'):
```

Comment:
do projects take _both_ of these fields??

Reply:
the Projects api has this field, and thinking now we also need to modify the JobTemplate, as it's the same case. What do you think @AlanCoding @kdelee?

Reply:
I believe it's on my to do list to modify unified_jobs

Reply:
My question is more basic here. Look at the help text for the two fields

`execution_environment`

> The container image to be used for execution.

`default_environment`

> The default execution environment for jobs run using this project.

What is the intent here? Is it that project updates will run in the `execution_environment` container? Because if it is, that's going to be a problem.

We don't allow customizing the custom virtual environment of project updates now (only that of jobs that use the project), and we won't allow customizing the execution environment. Our discussions of the OpenShift solution for project updates are completely incompatible with letting users select the EE to run a _project update_ in. This question is mostly directed at @jbradberry, but I also don't want to add defunct fields here because it indicates an intent to test them.

Reply:
I filed https://github.com/shanemcd/awx/pull/27

So this awxkit code should only need the `default_environment` field. However, the other field should not do any harm here.

Reply:
@yagomarques job templates do not have `default_environment`.

Reply:
@AlanCoding the `execution_environment` field is brought in by `Project` being a subclass of `UnifiedJobTemplate`.  If we want to prevent any kind of customization for project updates, we are going to have to override the EE resolver for projects.  Also, we're going to want to either prevent this field from showing up in the UI and API serializer, or we'll have to break up the usage of `ExecutableEnvironmentMixin` to the explicit job template and job types.  This might also have implications due to the field not being on the common parent table, I haven't thought that part through yet.

Reply:
> the execution_environment field is brought in by Project being a subclass of UnifiedJobTemplate. If we want to prevent any kind of customization for project updates, we are going to have to override the EE resolver for projects. Also, we're going to want to either prevent this field from showing up in the UI and API serializer

Agreed, this is what my linked PR does and I'm satisfied with the implications of it.

> or we'll have to break up the usage of ExecutableEnvironmentMixin to the explicit job template and job types

Or leave it where it is, and let it be `None` on the `Project` model and don't touch it. It could still be edited in the ORM, but you could lock down editing it if you were really worried.

There are other cases where a field on the UJT model does not apply to all job types.

Reply:
@AlanCoding What linked PR?  I don't see that you have a PR linked anywhere in here.

Reply:
https://github.com/shanemcd/awx/pull/27

Reply:
... why the heck did I think that was an issue instead of a PR?  :confused: